File: python/0513找树左下角的值.py
from binarytree import TreeNode

def findBottomLeftValue(root:TreeNode) -> int:
    # 最大深度、最后一层最左叶子值
    lst=[1,root.val]

    def _traversal(root:TreeNode,depth:int=1):
        if(root):
            # 只记录同一层最先进入的节点的值
            if(depth>lst[0]):
                lst[1]=root.val
                lst[0]=depth
            # 先进入左子树
            _traversal(root.left,depth+1)
            _traversal(root.right,depth+1)
    
    _traversal(root)

    return lst[1]

File: cpp/0513找树左下角的值.cpp
#include"mybitree.h"
#include<queue>

int findBottomLeftValue(TreeNode *root)
{   
    int res;
    queue<TreeNode *> nodes;
    nodes.push(root);

    while(!nodes.empty())
    {
        int size = nodes.size();

        for (int i = 0;i<size;++i)
        {
            TreeNode *crt = nodes.front();
            nodes.pop();
            if (i == 0)
                res = crt->val;
            
            if(crt->left)
                nodes.push(crt->left);
            if(crt->right)
                nodes.push(crt->right);
        }
    }

    return res;
}